packages/dds/legacy-dds/src/array/sharedArrayFactory.ts

```diff
@@ -53,8 +54,17 @@ export class SharedArrayFactory<T extends SerializableTypeForSharedArray>
 		/**
 		 * * The SharedArray
 		 */
-		const sharedArray = new SharedArray<T>(id, document, this.attributes);
+		const sharedArray = new SharedArrayClass<T>(id, document, this.attributes);
 		sharedArray.initializeLocal();
 		return sharedArray;
 	}
 }
+
+/**
+ * Entrypoint for {@link ISharedArray} creation.
+ * @legacy
+ * @alpha
+ */
+export const SharedArray = createSharedObjectKind<
+	ISharedArray<SerializableTypeForSharedArray>
```

Comment:
i don't know if we have another generic dds like this. we might actually want this to be a function.

``
export const SharedArray = <T extends SerializableTypeForSharedArray>()=>createSharedObjectKind<
	ISharedArray<T>(SharedArrayFactory<T>);

``

Reply:
Cell or ConsensusRegisterCollection are the closest analogs we have I think (generic with <T = any>, so not exactly the same).  Could maybe mimic those?